### ISSUE TITLE
Improved self redirect detection

### ISSUE THREAD

**Issue by ikreymer** [ikreymer/pywb#73]
Self-redirect loops may still occur due to url 'canonicalization', different urls get indexed as the same url.
Common example is http vs https and www. prefix vs without often get treated as same url, however other situations are possible. Leaving this open to track improvements to self-redirect.

**Comment by ikreymer** [comment]
To address self-redirect relating to http://www.somesite.example.com -> http://somesite.example.com redirect, add removal of `www[\d]*` prefix to `ReplayView.strip_scheme_www()`
Addresses a specific example of self-redirect in the most common url canonicalization scheme.

For example, in an index such as below, the two 301 requests will now be filtered out as they redirect to the same page. A request for 2015/http://example.com/ will return only the 200 response.

```
original time original status redirect
com,example)/ 2015 http://example.com/ 301 https://example.com/
com,example)/ 2015 https://example.com/ 301 https://www.example.com/
com,example)/ 2015 https://www.example.com/
```

Other approaches may be to actually follow the redirects, but detect redirect chain and also match to exact original url.

Will add other ideas to this issue.

**Comment by ikreymer** [comment]
Closing for now, as the original bug has been fixed in this case as part of 0.8.2
